manifests/bridge/debian.pp

```diff
@@ -0,0 +1,56 @@
+# = Define: network::bridge::debian
+#
+define network::bridge::debian(
+  $ports,
+  $ensure    = present,
+  $ipaddress = undef,
+  $netmask   = undef,
+  $method    = undef,
+  $family    = undef,
+  $onboot    = undef,
+  $mtu       = undef,
+
+  $ageing     = undef,
+  $bridgeprio = undef,
+  $fd         = undef,
+  $gcint      = undef,
+  $hello      = undef,
+  $hw         = undef,
+  $maxage     = undef,
+  $maxwait    = undef,
+  $pathcost   = undef,
+  $portprio   = undef,
+  $stp        = undef,
+  $waitport   = undef,  
+) {
+
+  $raw = {
+      'bridge_ports'      => join($ports, ' '),
+      'bridge_ageing'     => $ageing,
+      'bridge_bridgeprio' => $bridgeprio,
+      'bridge_fd'         => $fd,
+      'bridge_gcint'      => $gcint,
+      'bridge_hello'      => $hello,
+      'bridge_hw'         => $hw,
+      'bridge_maxage'     => $maxage,
+      'bridge_maxwait'    => $maxwait,
+      'bridge_pathcost'   => $pathcost,
+      'bridge_portprio'   => $portprio,
+      'bridge_stp'        => $stp,
+      'bridge_waitport'   => $waitport,
+  }
+
+  $opts = compact_hash($raw)
```

Comment:
It predates the function, I believe.
On Feb 16, 2015 9:42 AM, "Joshua Hoblitt" <notifications@github.com> wrote:

> In manifests/bridge/debian.pp
> <https://github.com/puppet-community/puppet-network/pull/85#discussion_r24765254>
> :
>
> > +      'bridge_ports'      => join($ports, ' '),
> > +      'bridge_ageing'     => $ageing,
> > +      'bridge_bridgeprio' => $bridgeprio,
> > +      'bridge_fd'         => $fd,
> > +      'bridge_gcint'      => $gcint,
> > +      'bridge_hello'      => $hello,
> > +      'bridge_hw'         => $hw,
> > +      'bridge_maxage'     => $maxage,
> > +      'bridge_maxwait'    => $maxwait,
> > +      'bridge_pathcost'   => $pathcost,
> > +      'bridge_portprio'   => $portprio,
> > +      'bridge_stp'        => $stp,
> > +      'bridge_waitport'   => $waitport,
> > +  }
> > +
> > +  $opts = compact_hash($raw)
>
> How is this function different from delete_undef_values() in stdlib?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/puppet-community/puppet-network/pull/85/files#r24765254>
> .
>